common/s2n-codec/src/decoder/checked_range.rs

```diff
@@ -12,9 +12,26 @@ pub struct CheckedRange {
     original_ptr: *const u8,
 }
 
+#[cfg(any(test, feature = "generator"))]
+use bolero_generator::*;
+
+#[cfg(test)]
+impl bolero::TypeGenerator for CheckedRange {
+    fn generate<D: bolero::Driver>(driver: &mut D) -> Option<Self> {
+        let start = gen::<usize>().generate(driver)?;
+        let end = (start..).generate(driver)?;
+        Some(CheckedRange { start, end })
+    }
+}
+
 impl CheckedRange {
     #[inline]
     pub(crate) fn new(start: usize, end: usize, original_ptr: *const u8) -> Self {
+        debug_assert!(
+            end >= start,
+            "{}",
+            format!("end: {end} must be greater than or equal to start: {start}")
```

Comment:
Oh you don't need `format` for this
```suggestion
            "end: {end} must be greater than or equal to start: {start}"
```